harness/typeCoercion.js

```diff
@@ -0,0 +1,254 @@
+// Copyright (C) 2017 Josh Wolfe. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+description: |
+    Functions to help generate test cases for testing type coercion abstract
+    operations like ToNumber.
+---*/
+
+function getValuesCoercibleToIntegerZero() {
+  var result = [];
+
+  var primitiveValues = [
+    // ToNumber
+    null,
+    false,
+    0,
+    "0",
+
+    // ToInteger: NaN -> +0
+    undefined,
+    NaN,
+    "",
+    "foo",
+    "true",
+
+    // ToInteger: floor(abs(number))
+    0.9,
+    -0,
+    -0.9,
+    "0.9",
+    "-0",
+    "-0.9",
+  ];
```

Comment:
I don't quite understand the phrasing here. A particular test which uses this function will then use the values for a particular parameter, which is cast in a particular way. Wouldn't it make more sense to have distinct functions for getting values that coerce to zero via ToInteger, via ToLength, etc? Seems like this function implements the ToInteger one (should that figure into the name?); for ToLength, for example, you could also include -10, and for ToNumber, you would remove many of these.

Reply:
Does the latest commit clarify the situation or make it more obscure? https://github.com/tc39/test262/pull/1200/commits/2bcee866c2f75796403a3024792ad493752ae2c0